sdk/keyvault/keyvault-admin/review/keyvault-admin.api.md

```diff
@@ -166,6 +197,21 @@ export interface KeyVaultSelectiveKeyRestoreResult {
     startTime: Date;
 }
 
+// @public
+export type KeyVaultSetting = BooleanKeyVaultSetting | DefaultKeyVaultSetting;
+
+// @public
+export class KeyVaultSettingsClient {
+    constructor(vaultUrl: string, credential: TokenCredential, options?: SettingsClientOptions);
+    getSetting(settingName: string, options: GetSettingOptions): Promise<KeyVaultSetting>;
+    getSettings(options: GetSettingsOptions): Promise<GetSettingsResponse>;
+    updateSetting(settingName: string, value: string, options: CreateOrUpdateSettingOptions): Promise<KeyVaultSetting>;
```

Comment:
The pattern here has changed, though be idiomatic. `KeyVaultSetting` is now mutable, and you'd pass that to `updateSetting` so that all type conversions are handled therein. Also, we don't really want users to have to serialize a `boolean` themselves e.g., should it be "true" or "True"? At the very least, `value` should be `boolean` which you could change to a union later. At least for .NET, we made the decision not so support arbitrary strings for now (no escape hatch).

/cc @vcolin7 for what Java decided to do in this regard.

Reply:
I've changed it now so that KeyVaultSetting is a simple interface with a boolean `value`. My concern with this approach is that adding multiple setting types in future could be a breaking change. As is, `KeyVaultSetting` is used as a return type, and it wouldn't be guaranteed that the setting value returned is a boolean anymore if we were to change `value` to be a union, so you would get a compile error. But we might be able to get around that by leveraging generics instead of a union.

Reply:
Think I might have misunderstood this one. I have changed `updateSetting` to take in a `boolean`.